src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -315,31 +437,94 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
+            ReturnErrorOnFailure(global_scene_count.Load(storage));
+            global_scene_count.count_value--;
+            ReturnErrorOnFailure(global_scene_count.Save(storage));
+
             scene_count--;
             scene_map[scene.index].Clear();
-            ReturnErrorOnFailure(this->Save(storage));
+            err = this->Save(storage);
+
+            if (err != CHIP_NO_ERROR)
+            {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+            }
 
             err = scene.Delete(storage);
 
             // On failure to delete scene, undoes the change to the Fabric Scene Data
             if (err != CHIP_NO_ERROR)
             {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+
                 scene_count++;
                 scene_map[scene.index] = scene.mStorageId;
                 ReturnErrorOnFailure(this->Save(storage));
             }
         }
         return err;
     }
+
+    CHIP_ERROR Load(PersistentStorageDelegate * storage) override
+    {
+        VerifyOrReturnError(nullptr != storage, CHIP_ERROR_INVALID_ARGUMENT);
+
+        uint8_t buffer[kPersistentFabricBufferMax] = { 0 };
+        StorageKeyName key                         = StorageKeyName::Uninitialized();
+
+        // Set data to defaults
+        Clear();
+
+        // Update storage key
+        ReturnErrorOnFailure(UpdateKey(key));
+
+        // Load the serialized data
+        uint16_t size  = static_cast<uint16_t>(sizeof(buffer));
+        CHIP_ERROR err = storage->SyncGetKeyValue(key.KeyName(), buffer, size);
+        VerifyOrReturnError(CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND != err, CHIP_ERROR_NOT_FOUND);
+        ReturnErrorOnFailure(err);
+
+        // Decode serialized data
+        TLV::TLVReader reader;
+        reader.Init(buffer, size);
+
+        err = Deserialize(reader);
+        // If PersistentData::Load returns "buffer too small", the table in flash memory holds to many scenes (Can happend if
+        // max_scenes_per_fabric was reduced during an OTA). The fabric will then proceed to remove all scenes above the limit from
+        // the fabric
+        if (err == CHIP_ERROR_BUFFER_TOO_SMALL)
+        {
+            reader.Init(buffer, size);
+            ReturnErrorOnFailure(DeserializeAjust(reader, storage));
+            return this->Save(storage);
+        }
+
+        return err;
+    }
 };
 
 CHIP_ERROR DefaultSceneTableImpl::Init(PersistentStorageDelegate * storage)
 {
+    GlobalSceneCount global_count;
+
     if (storage == nullptr)
     {
         return CHIP_ERROR_INCORRECT_STATE;
     }
 
+    CHIP_ERROR err = global_count.Load(storage);
+    VerifyOrReturnError(CHIP_NO_ERROR == err || CHIP_ERROR_NOT_FOUND == err, err);
+
+    if (err == CHIP_ERROR_NOT_FOUND)
+    {
+        global_count.count_value = 0;
+        global_count.Save(storage);
```

Comment:
Do we need to have a global_count save? Can't we assume when it is "loaded" that if global_count isn't found, the value implicitly is 0